File: C/delete_node_in_a_Linked_List.c
/* Write a function to delete a node (except the tail) in a singly linked
   list, given only access to that node. */

   /**
   * Definition for singly-linked list.
   * struct ListNode {
   *     int val;
   *     struct ListNode *next;
   * };
   */
void deleteNode(struct ListNode* node) {
  if (node == NULL || node -> next == NULL) {
    return;
  }
  node->val = node->next->val;
  node->next = node->next->next;
}

File: Python/delete_node_in_a_Linked_List.py
""" Write a function to delete a node (except the tail) in a singly linked 
    list, given only access to that node. """
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteNode(self, node):
        if node is None or node.next is None:
            delete_node = node.next
            node.val = delete_node.val
            node.next = delete_node.next
            del delete_node
